### ISSUE TITLE
composer 1.7 doesn't work with `wp-cli/wp-cli` package

### ISSUE THREAD

**Issue by evilebottnawi** [composer/composer#7516]
My `composer.json`:

```json
{
    "name": "test/test",
    "type": "project",
    "license": "MIT",
    "require-dev": {
        "wp-cli/wp-cli": "~1.5.0",
    }
}

```

Output of `composer diagnose`:

```
Checking composer.json: OK
Checking platform settings: OK
Checking git settings: OK
Checking http connectivity to packagist: OK
Checking https connectivity to packagist: OK
Checking github.com rate limit: OK
Checking disk free space: OK
Checking pubkeys: 
Tags Public Key Fingerprint: 57815BA2 7E54DC31 7ECC7CC5 573090D0  87719BA6 8F3BB723 4E5D42D0 84A14642
Dev Public Key Fingerprint: 4AC45767 E5EC2265 2F0C1167 CBBB8A2B  0C708369 153E328C AD90147D AFE50952
OK
Checking composer version: OK
Composer version: 1.7.0
PHP version: 7.2.8
PHP binary path: /usr/bin/php7.2
```

When I run this command:

```
vendor/bin/wp
```

I get the following output:

```
PHP Fatal error:  Class 'WP_CLI\Loggers\Regular' not found in /path/to/vendor/wp-cli/wp-cli/php/WP_CLI/Loggers/Execution.php on line 8

```

And I expected this to happen:

No errors, `1.6.5` version works without errors

**Comment by alcohol** [comment]
Cannot reproduce.

**Comment by wojsmol** [comment]
@alcohol Did there was same changes to created autoloaders between 1.6.5 and 1.7.0?

**Comment by Seldaek** [comment]
@localheinz You can temporarily use `composer self-update --snapshot` to get the fix, I'll release a tag as soon as #7517 gets fixed which should be soon enough.

**Comment by localheinz** [comment]
Awesome, thanks a ton, @seldaek!
